### DISCUSSION TITLE
Wanting to use on a VPS with a Tailscale tunnel to my LAN. How can I set this up to send the clients real IP to the servers on my LAN?

### DISCUSSION THREAD

**Discussion by relink2013** [ZoeyVid/NPMplus#776] [Q&A] [answered]
I am currently using the official NPM on a VPS, this then routes through a Tailscale tunnel to my lan. I have Tailscale setup so that I can use my actual LAN IP addresses instead of the Tailscale addresses. 

The problem is that all of my servers only see the IP address of the local Tailscale instance and not the real IP seen by the reverse proxy. This is making setting up proper security a nightmare. 

I’d like to switch to NPMplus and in the process I’d really like to fix this issue of not seeing the real IP addresses. How could I do this? And preferably without needing to adjust any of the servers on my LAN?

**Comment by Zoey2936** [comment, marked as answer]
Do your backwnd servers read the forworded for/x real ip header?

**Reply by relink2013** [reply]
Not that I ever setup. Some of them like my Synology aren't as easy to modify things like that on. 

I know I used to have the exact same setup before but I used Wireguard to connect OPNsense to the VPS and it worked perfectly fine. I'm not sure what changed in the switch that would cause this.

**Reply by Zoey2936** [reply]
they can only now the "real" ip if they use these headers sent by NPMplus
